src/sotauptaneclient.cc

```diff
@@ -21,8 +21,8 @@ void SotaUptaneClient::run(command::Channel *commands_channel) {
 }
 
 bool SotaUptaneClient::isInstalled(const Uptane::Target &target) {
-  if (target.ecu_identifier() == config.uptane.primary_ecu_serial) {
-    return target.filename() == OstreePackage::getEcu(config.uptane.primary_ecu_serial, config.ostree.sysroot).ref_name;
+  if (target.ecu_identifier() == uptane_repo.primary_ecu_serial) {
+    return target.filename() == OstreePackage::getEcu(uptane_repo.primary_ecu_serial, config.ostree.sysroot).ref_name;
```

Comment:
Sounds like the real fix is harder than I'd guessed/hoped for :)  Just adding a getter isn't really going to help us get toward the goal, so I agree that fiddling with friend's privates here is OK.